src/System.ComponentModel.TypeConverter/src/System/misc.cs

```diff
@@ -0,0 +1,191 @@
+        {
+            get
+            {
+                if (!_replaced)
+                {
+                    _replaced = true;
+                    base.DescriptionValue = SR.GetResourceString(base.Description, "");
+                }
+                return base.Description;
+            }
+        }
+
+        public SRDescriptionAttribute(string description) : base(description)
+        {
+        }
+    }
+
+    [AttributeUsage(AttributeTargets.All)]
+    internal sealed class SRCategoryAttribute : CategoryAttribute
+    {
+        public SRCategoryAttribute(string category) : base(category)
+        {
+        }
+
+        protected override string GetLocalizedString(string value)
+        {
+            return SR.GetResourceString(value, "");
+        }
+    }
+
+
+    // from Misc/SecurityUtils.cs
+
+    internal static class SecurityUtils
+    {
+        /// <summary>
+        ///     This helper method provides safe access to Activator.CreateInstance.
+        ///     NOTE: This overload will work only with public .ctors. 
+        /// </summary>
+        internal static object SecureCreateInstance(Type type)
+        {
+            return SecureCreateInstance(type, null);
+        }
+
+        /// <summary>
+        ///     This helper method provides safe access to Activator.CreateInstance.
+        ///     Set allowNonPublic to true if you want non public ctors to be used. 
+        /// </summary>
+        internal static object SecureCreateInstance(Type type, object[] args)
+        {
+            if (type == null)
+            {
+                throw new ArgumentNullException("type");
+            }
+
+            //BindingFlags flags = BindingFlags.Instance | BindingFlags.Public | BindingFlags.CreateInstance;
+
+            // // if it's an internal type, we demand reflection permission.
+            // if (!type.IsVisible) {
+            //     DemandReflectionAccess(type);
+            // }
+            // else if (allowNonPublic && !HasReflectionPermission(type)) {
+            //     // Someone is trying to instantiate a public type in *our* assembly, but does not
+            //     // have full reflection permission. We shouldn't pass BindingFlags.NonPublic in this case.
+            //     // The reason we don't directly demand the permission here is because we don't know whether
+            //     // a public or non-public .ctor will be invoked. We want to allow the public .ctor case to
+            //     // succeed.
+            //     allowNonPublic = false;
+            // }
+
+            // if (allowNonPublic) {
+            //     flags |= BindingFlags.NonPublic;
+            // }
+
+            return Activator.CreateInstance(type, args);
+        }
+    }
+
+    internal static class HResults
+    {
+        internal const int License = unchecked((int)0x80131901);
+    }
+}
+
+namespace System.Resources
+{
+    public partial interface IResourceReader : System.Collections.IEnumerable, System.IDisposable
```

Comment:
Why does this need to be here? Can it not just reference System.Resources.Reader?